core/lib/libtomcrypt/dh.c

```diff
@@ -45,6 +45,9 @@ TEE_Result crypto_acipher_gen_dh_key(struct dh_keypair *key, struct bignum *q,
 	if (key_size != 8 * mp_unsigned_bin_size(key->p))
 		return TEE_ERROR_BAD_PARAMETERS;
 
+	if (mp_prime_is_prime(key->p, 0, &ltc_res) != CRYPT_OK || ltc_res != LTC_MP_YES)
```

Comment:
A primality test can take significant time especially when the value is indeed a prime number which is the normal case here. I think it would be better to deal with the error later when it is detected before it causes a crash, and return an error code to the TA.

Reply:
I will look into it. Thank you Jérôme.